setup.cfg

```diff
@@ -34,6 +34,7 @@ omit =
 	*/signac/common/configobj/*.py
 
 [tool:pytest]
+xfail_strict = True
```

Comment:
I consider this a rather drastic change in the behavior of the overall test suite. Should we consider to apply this a bit more granular, i.e., use `strict=True` where this is relevant?

Reply:
Actually, wanting to enable this was my original goal and what made me remember that these multithreaded tests had been xfailed. IMO setting this can only be a good thing because it forces us to immediately deal with changes that incidentally and unexpectedly fix a bug or change behaviors in other unexpected ways that we only catch because a test suddenly starts xpassing. The value of `strict` passed to the decorator supersedes the value in the file, so I would rather have `xfail_strict=True` configured here and then manually mark with `strict=False` cases where for whatever reason we actually need to allow xfails, for instance because a test is flaky. I would prefer to force developers to jump through hoops in order to enable flaky tests.

Reply:
I'm +1 for making the change at a global level, but I would be okay with doing it in a separate PR, or at least adding a separate changelog line to indicate this.

Reply:
No need for separate PR IMO (albeit cleaner), but +1 for dedicated changelog entry.